pages/index.js

```diff
@@ -57,103 +66,204 @@ class Index extends React.Component {
     })
   }
 
+  showCreateQueuePanel() {
+    this.setState({
+      showCreateQueuePanel: true,
+    })
+  }
+
+  hideCreateQueuePanel() {
+    this.setState({
+      showCreateQueuePanel: false,
+    })
+  }
+
   createCourse(course) {
     this.props.createCourse(course).then(() => this.hideCreateCoursePanel())
   }
 
+  createQueue(queue, courseId) {
+    this.props.createQueue(courseId, queue).then(() => this.hideCreateQueuePanel())
+  }
+
+  deleteQueue(courseId, queueId) {
+    this.props.deleteQueue(courseId, queueId)
```

Comment:
is it possible to have this pop up a modal (similar to components/ConfirmLeaveQueueModal.js ) in order to help prevent accidental deletion?

Reply:
yep - that was already on my todo list for another PR, but that makes sense to do here.